bridges/CNETBridge.php

```diff
@@ -0,0 +1,109 @@
+		$article_html = str_replace(' height="0"', '', $article_html);
+		$article_html = str_replace('<noscript>', '', $article_html);
+		$article_html = str_replace('</noscript>', '', $article_html);
+		$article_html = StripWithDelimiters($article_html, '<a class="clickToEnlarge', '</a>');
+		$article_html = stripWithDelimiters($article_html, '<span class="nowPlaying', '</span>');
+		$article_html = stripWithDelimiters($article_html, '<span class="duration', '</span>');
+		$article_html = stripWithDelimiters($article_html, '<script', '</script>');
+		$article_html = stripWithDelimiters($article_html, '<svg', '</svg>');
+		return $article_html;
+	}
+
+	public function collectData() {
+
+		// Retrieve and check user input
+		$topic = str_replace('-', '/', $this->getInput('topic'));
+		if (!empty($topic) && (substr_count($topic, '/') > 1 || !ctype_alpha(str_replace('/', '', $topic))))
+			returnClientError('Invalid topic: ' . $topic);
+
+		// Retrieve webpage
+		$pageUrl = self::URI . (empty($topic) ? 'news/' : $topic.'/');
+		$html = getSimpleHTMLDOM($pageUrl)
+		or returnServerError('Could not request CNET: '.$pageUrl);
+		$limit = 0;
+
+		// Process articles
+		foreach($html->find('div.assetBody, div.riverPost') as $element) {
+			if ($limit < 10) {
+
+				$article_title = trim($element->find('h2, h3', 0)->plaintext);
+				$article_uri = self::URI . (substr($element->find('a', 0)->href, 1));
+				$article_thumbnail = $element->parent()->find('img[src]', 0)->src;
+				$article_timestamp = strtotime($element->find('time.assetTime, div.timeAgo', 0)->plaintext);
+				$article_author = trim($element->find('a[rel=author], a.name', 0)->plaintext);
+				$article_content = '<p><b>' . trim($element->find('p.dek', 0)->plaintext) . '</b></p>';
+
+				if (is_null($article_thumbnail))
+					$article_thumbnail = extractFromDelimiters($element->innertext, '<img src="', '"');
+
+				if (!empty($article_title) && !empty($article_uri) && strpos($article_uri, self::URI . 'news/') !== false) {
+
+					$article_html = getSimpleHTMLDOMCached($article_uri) or $article_html = null;
+
+					if (!is_null($article_html)) {
+
+						if (empty($article_thumbnail))
+							$article_thumbnail = $article_html->find('div.originalImage', 0);
+						if (empty($article_thumbnail))
+							$article_thumbnail = $article_html->find('span.imageContainer', 0);
+						if (is_object($article_thumbnail))
+							$article_thumbnail = $article_thumbnail->find('img', 0)->src;
+
+						$article_content .= trim(
+							$this->cleanArticle(
+								extractFromDelimiters(
+									$article_html, '<article', '<footer'
+								)
+							)
+						);
+					}
+
+					$item = array();
+					$item['uri'] = $article_uri;
+					$item['title'] = $article_title;
+					$item['author'] = $article_author;
+					$item['timestamp'] = $article_timestamp;
+					$item['enclosures'] = array($article_thumbnail);
+					$item['content'] = $article_content;
+					$this->items[] = $item;
+					$limit++;
```

Comment:
Same as before, the limit should be just after line 62

Reply:
In the case of this bridge, yes it's possible since 1 checked item = 1 processed item. Same as before, I think there is no `continue` or similar statement that could skip increasing the limit, but it's safer to increase it as soon as possible, indeed. I'll fix that.

Reply:
You can get rid of the if condition by simply checking the array size and breaking the loop:

```PHP
if(count($this->items) >= 10) {
    break;
}
```

Reply:
Thanks, changing to that. :+1: